internal/clients/gitdiff/gitdiff.go

```diff
@@ -0,0 +1,66 @@
+package gitdiff
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+const unknownCommitHash = "0000000000000000000000000000000000000000"
+
+// GitDiff client for getting diffs from the command line
+type GitDiff struct {
+	WorkDir    string
+	BaseBranch string
+	BaseSHA    string
+	Head       string
+}
+
+// GetDiff uses the command line to compute the diff
+func (gd *GitDiff) GetDiff() (string, error) {
+	err := os.Chdir(gd.WorkDir)
+	if err != nil {
+		return "", err
+	}
+
+	var diffCmd *exec.Cmd
+	if gd.BaseBranch != "" {
+		err = exec.Command("git", "fetch", "origin", gd.BaseBranch, "--depth=1").Run()
+		if err != nil {
+			return "", err
+		}
+		diffCmd = exec.Command("git", "diff", fmt.Sprintf("origin/%s", gd.BaseBranch))
+	} else if gd.BaseSHA == "" || gd.BaseSHA == unknownCommitHash {
+		err = exec.Command("git", "fetch", "origin", gd.Head, "--depth=2").Run()
```

Comment:
Setting depth=2 because if the actions is running on the push of a new branch (just added to remote) we only want to get the diff of its first commit. If depth=1 then the entire repo is stored at the first commit and thus we would end up getting the entire repo as the diff.

Reply:
Aside from the initial push to an new empty repository, would there be other instances that would enter this condition block? 

I'm curious as to whether the BaseSHA can be ""/`unknownCommitHash` on the push of a new branch for a repo with existing commits on the `master` branch, since this condition was initially added just for the purpose of the first push to an empty repo.

Reply:
On the first push for any new branch will have the `""/unknownCommitHash`, since the action has no before commit for it.